### ISSUE TITLE
add_host always triggers, ignores conditionals

### ISSUE THREAD

**Issue by omgjlk** [ansible/ansible#2445]
I've setup a simple test that has a conditional of with_string to execute tasks.  I can successfully avoid a task execution of debug, but not of add_host.  Here is the contents of my playbook:

---
- name: host-tester
  hosts: 127.0.0.1
  connection: local
  gather_facts: false
  
  tasks:
  - name: debug
    action: debug msg="True var is ${true}"
  - name: debug2
    action: debug msg="Second debug message"
    when_string: ${truevar} == 'True'
  - name: add_host
    action: add_host hostname='mynewhost'
    when_string: ${truevar} == 'True'

and here is the execution examples:

$ ansible-playbook -vvv -i hosts host-test.yml -e truevar=True

PLAY [host-tester] *********************

TASK: [debug] *********************
ok: [127.0.0.1] => {"msg": "True var is ${true}"}

TASK: [debug2] *********************
ok: [127.0.0.1] => {"msg": "Second debug message"}

TASK: [add_host] *********************
creating host via 'add_host': hostname=mynewhost
changed: [127.0.0.1] => {"changed": true, "new_host": "mynewhost"}

PLAY RECAP *********************
127.0.0.1                      : ok=3    changed=1    unreachable=0    failed=0

$ ansible-playbook -vvv -i hosts host-test.yml -e truevar=False

PLAY [host-tester] *********************

TASK: [debug] *********************
ok: [127.0.0.1] => {"msg": "True var is ${true}"}

TASK: [debug2] *********************
skipping: [127.0.0.1]

TASK: [add_host] *********************
creating host via 'add_host': hostname=mynewhost
changed: [127.0.0.1] => {"changed": true, "new_host": "mynewhost"}

PLAY RECAP *********************
127.0.0.1                      : ok=2    changed=1    unreachable=0    failed=0

As you can see the add_host task was ran both times, regardless of the value of truevar.

**Comment by omgjlk** [comment]
Lets try that paste again but with proper formatting.

```
---
- name: host-tester
  hosts: 127.0.0.1
  connection: local
  gather_facts: false

  tasks:
    - name: debug
      action: debug msg="True var is ${true}"

    - name: debug2
      action: debug msg="Second debug message"
      when_string: ${truevar} == 'True'

    - name: add_host
      action: add_host hostname='mynewhost'
      when_string: ${truevar} == 'True'
```

**Comment by mpdehaan** [comment]
So, just looked and add_hosts is a "bypass host loop" plugin, that means regardless of how many hosts you have, which is why conditionals aren't working.

I'm currently looking to see if there is a nice way to make them work :)

**Comment by mpdehaan** [comment]
Ok, that was easy, it was turned off for no good reason :)

Fixed in 7044e53
